File: app/src/main/java/org/bluechat/blueninemenmoris/model/Actor.java
package org.bluechat.blueninemenmoris.model;

/**
 * Created by Samsaini on 05/25/2016.
 */
public class Actor {
    //things an actor can have
    //position
    private int posx;
    private int posy;
    private int prex;
    private int prey;
    private int rmx;
    private int rmy;
    private boolean removed;
    private boolean availableToRemove;
    private boolean placed;
    private int placedIndex = -1;
    //Player that have this Actor
    private int player;
    private int number;


    public Actor(int posx, int posy, int number) {
        this.posx = posx;
        this.posy = posy;
        this.prex = posx;
        this.prey = posy;
        this.number = number;
        placed = false;
    }

    /*
     *Get player
     */
    public int getPlayer() {
        return player;
    }

    public void setPlayer(int player) {
        this.player = player;
    }

    public int getPosy() {
        return posy;
    }

    public void setPosxy(int posx, int posy) {
        this.posx = posx;
        this.posy = posy;
    }
    public void setToPreviousPosition(){
        posx = prex;
        posy = prey;
    }
    public int getPosx() {
        return posx;
    }

    public int getPrePosy() {
        return prey;
    }
    public int getPrePosx() {
        return prex;
    }

    public void setPrePosxy(int prex, int prey) {
        this.prex = prex;
        this.prey = prey;
    }


    public boolean isRemoved() {
        return removed;
    }

    public void setRemoved(boolean removed) {
        this.removed = removed;
        this.setPlacedIndex(111);
        this.placed = false;
    }

    public boolean isAvailableToRemove() {
        return availableToRemove;
    }

    public void setAvailableToRemove(boolean availableToRemove) {
        this.availableToRemove = availableToRemove;
    }

    public boolean isPlaced() {
        return placed;
    }

    public void setPlaced(boolean placed) {
        this.placed = placed;
    }

    public int getPlacedIndex() {
        return placedIndex;
    }

    public void setPlacedIndex(int placedIndex) {
        this.placed = true;
        this.placedIndex = placedIndex;
    }

    public int getNumber() {
        return number;
    }

    public void setNumber(int number) {
        this.number = number;
    }
}

File: app/src/main/java/org/bluechat/blueninemenmoris/model/Actor.kt
package org.bluechat.blueninemenmoris.model

/**
 * Created by Samsaini on 05/25/2016.
 */
class Actor(posx: Int, posy: Int, var number: Int) {
    //things an actor can have
    //position
    var posx: Int = 0
        private set
    var posy: Int = 0
        private set
    var prePosx: Int = 0
        private set
    var prePosy: Int = 0
        private set
    private val rmx: Int = 0
    private val rmy: Int = 0
    var isRemoved: Boolean = false
        set(removed) {
            field = removed
            this.placedIndex = 111
            this.isPlaced = false
        }
    var isAvailableToRemove: Boolean = false
    var isPlaced: Boolean = false
    var placedIndex = -1
        set(placedIndex) {
            this.isPlaced = true
            field = placedIndex
        }
    //Player that have this Actor
    /*
     *Get player
     */
    var player: Int = 0


    init {
        this.posx = posx
        this.posy = posy
        this.prePosx = posx
        this.prePosy = posy
        isPlaced = false
    }

    fun setPosxy(posx: Int, posy: Int) {
        this.posx = posx
        this.posy = posy
    }

    fun setToPreviousPosition() {
        posx = prePosx
        posy = prePosy
    }

    fun setPrePosxy(prex: Int, prey: Int) {
        this.prePosx = prex
        this.prePosy = prey
    }
}
